// angularx-score/esm2022/lib/shared/components/org-tree/org-tree.component.mjs
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "primeng/dynamicdialog";
import * as i2 from "primeng/tree";
export class OrgTreeComponent {
    constructor(_injector, dialogService) {
        this._injector = _injector;
        this.dialogService = dialogService;
        this.onSelectNode = new EventEmitter();
        this.loading = false;
    }
    ngAfterViewInit() {
        this.loading = true;
        this.orgService.initOrgTree().subscribe((res) => {
            this.orgTree = res?.data;
            this.loading = false;
        });
    }
    nodeExpand(event) {
        if (!!event.node.children) {
            return;
        }
        if (event.node) {
            this.loading = true;
            setTimeout(() => {
                this.orgService.getChildOrg(event.node.key).subscribe((res) => {
                    event.node.children = res.data;
                });
                this.loading = false;
                // this.cd.markForCheck();
            }, 200);
        }
    }
    nodeSelect(event) {
        this.onSelectNode.emit(event.node);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: OrgTreeComponent, deps: [{ token: i0.Injector }, { token: i1.DialogService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.5", type: OrgTreeComponent, selector: "app-org-tree", inputs: { orgService: "orgService" }, outputs: { onSelectNode: "onSelectNode" }, ngImport: i0, template: "<p-tree class=\"w-full md:w-30rem\" [value]=\"orgTree\" selectionMode=\"single\" [loading]=\"loading\"\n    (onNodeExpand)=\"nodeExpand($event)\" (onNodeSelect)=\"nodeSelect($event)\"></p-tree>", styles: ["::ng-deep app-org-tree .p-tree{border:none;padding:0}\n"], dependencies: [{ kind: "component", type: i2.Tree, selector: "p-tree", inputs: ["value", "selectionMode", "selection", "style", "styleClass", "contextMenu", "layout", "draggableScope", "droppableScope", "draggableNodes", "droppableNodes", "metaKeySelection", "propagateSelectionUp", "propagateSelectionDown", "loading", "loadingIcon", "emptyMessage", "ariaLabel", "togglerAriaLabel", "ariaLabelledBy", "validateDrop", "filter", "filterBy", "filterMode", "filterPlaceholder", "filteredNodes", "filterLocale", "scrollHeight", "lazy", "virtualScroll", "virtualScrollItemSize", "virtualScrollOptions", "indentation", "_templateMap", "trackBy", "virtualNodeHeight"], outputs: ["selectionChange", "onNodeSelect", "onNodeUnselect", "onNodeExpand", "onNodeCollapse", "onNodeContextMenuSelect", "onNodeDrop", "onLazyLoad", "onScroll", "onScrollIndexChange", "onFilter"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: OrgTreeComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-org-tree', template: "<p-tree class=\"w-full md:w-30rem\" [value]=\"orgTree\" selectionMode=\"single\" [loading]=\"loading\"\n    (onNodeExpand)=\"nodeExpand($event)\" (onNodeSelect)=\"nodeSelect($event)\"></p-tree>", styles: ["::ng-deep app-org-tree .p-tree{border:none;padding:0}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.Injector }, { type: i1.DialogService }]; }, propDecorators: { orgService: [{
                type: Input
            }], onSelectNode: [{
                type: Output
            }] } });
