// angular-l10n/esm2022/lib/directives/l10n-number.directive.mjs
import { Directive, Input } from '@angular/core';
import { L10nDirective } from '../models/l10n-directive';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nNumberDirective extends L10nDirective {
    set l10nNumber(options) {
        if (options)
            this.options = options;
    }
    constructor(intl) {
        super();
        this.intl = intl;
    }
    getValue(text) {
        return this.intl.formatNumber(text, this.options, this.language, this.currency, this.convert, this.convertParams);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberDirective, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nNumberDirective, isStandalone: true, selector: "[l10nNumber]", inputs: { l10nNumber: "l10nNumber", options: "options", currency: "currency", convert: "convert", convertParams: "convertParams" }, usesInheritance: true, ngImport: i0 }); }
}
export { L10nNumberDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nNumber]',
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; }, propDecorators: { l10nNumber: [{
                type: Input
            }], options: [{
                type: Input
            }], currency: [{
                type: Input
            }], convert: [{
                type: Input
            }], convertParams: [{
                type: Input
            }] } });
