// angular-l10n/esm2022/lib/directives/l10n-validate-number.directive.mjs
import { Directive, forwardRef, Input } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-validation";
/**
 * Function that takes a control and returns either null when it’s valid, or an error object if it’s not.
 * @param validation The instance of L10nValidation service
 * @param options A L10n or Intl NumberFormatOptions object
 * @param minValue The minimum value
 * @param maxValue The maximum value
 * @param language The current language
 * @return An error object: 'format', 'minValue' or 'maxValue'; null in case the value is valid
 */
export function l10nValidateNumber(validation, options, minValue = Number.MIN_VALUE, maxValue = Number.MAX_VALUE, language) {
    const validator = (c) => {
        if (c.value === '' || c.value == null)
            return null;
        const value = validation.parseNumber(c.value, options, language);
        if (value != null) {
            if (value < minValue) {
                return { minValue: true };
            }
            else if (value > maxValue) {
                return { maxValue: true };
            }
            return null; // The number is valid.
        }
        else {
            return { format: true };
        }
    };
    return validator;
}
class L10nValidateNumberDirective {
    set l10nValidateNumber(options) {
        if (options)
            this.options = options;
    }
    constructor(validation) {
        this.validation = validation;
    }
    ngOnInit() {
        this.validator = l10nValidateNumber(this.validation, this.options, this.minValue, this.maxValue, this.language);
    }
    ngOnChanges() {
        this.validator = l10nValidateNumber(this.validation, this.options, this.minValue, this.maxValue, this.language);
    }
    validate(c) {
        return this.validator ? this.validator(c) : null;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidateNumberDirective, deps: [{ token: i1.L10nValidation }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nValidateNumberDirective, isStandalone: true, selector: "[l10nValidateNumber][ngModel],[l10nValidateNumber][formControl],[l10nValidateNumber][formControlName]", inputs: { l10nValidateNumber: "l10nValidateNumber", options: "options", minValue: "minValue", maxValue: "maxValue", language: "language" }, providers: [
            { provide: NG_VALIDATORS, useExisting: forwardRef(() => L10nValidateNumberDirective), multi: true }
        ], usesOnChanges: true, ngImport: i0 }); }
}
export { L10nValidateNumberDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidateNumberDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nValidateNumber][ngModel],[l10nValidateNumber][formControl],[l10nValidateNumber][formControlName]',
                    providers: [
                        { provide: NG_VALIDATORS, useExisting: forwardRef(() => L10nValidateNumberDirective), multi: true }
                    ],
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nValidation }]; }, propDecorators: { l10nValidateNumber: [{
                type: Input
            }], options: [{
                type: Input
            }], minValue: [{
                type: Input
            }], maxValue: [{
                type: Input
            }], language: [{
                type: Input
            }] } });
