// angular-l10n/esm2022/lib/directives/l10n-date.directive.mjs
import { Directive, Input } from '@angular/core';
import { L10nDirective } from '../models/l10n-directive';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nDateDirective extends L10nDirective {
    set l10nDate(options) {
        if (options)
            this.options = options;
    }
    constructor(intl) {
        super();
        this.intl = intl;
    }
    getValue(text) {
        return this.intl.formatDate(text, this.options, this.language, this.timezone);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDateDirective, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nDateDirective, isStandalone: true, selector: "[l10nDate]", inputs: { l10nDate: "l10nDate", options: "options", timezone: "timezone" }, usesInheritance: true, ngImport: i0 }); }
}
export { L10nDateDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDateDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nDate]',
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; }, propDecorators: { l10nDate: [{
                type: Input
            }], options: [{
                type: Input
            }], timezone: [{
                type: Input
            }] } });
