// angular-l10n/esm2022/lib/directives/l10n-plural.directive.mjs
import { Directive, Input } from '@angular/core';
import { L10nDirective } from '../models/l10n-directive';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nPluralDirective extends L10nDirective {
    set l10nPlural(options) {
        if (options)
            this.options = options;
    }
    constructor(intl) {
        super();
        this.intl = intl;
    }
    getValue(text) {
        return this.intl.plural(text, this.prefix, this.options, this.language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralDirective, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nPluralDirective, isStandalone: true, selector: "[l10nPlural]", inputs: { l10nPlural: "l10nPlural", prefix: "prefix", options: "options" }, usesInheritance: true, ngImport: i0 }); }
}
export { L10nPluralDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nPlural]',
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; }, propDecorators: { l10nPlural: [{
                type: Input
            }], prefix: [{
                type: Input
            }], options: [{
                type: Input
            }] } });
