// angular-l10n/esm2022/lib/directives/l10n-validate-date.directive.mjs
import { Directive, forwardRef, Input } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-validation";
/**
 * Function that takes a control and returns either null when it’s valid, or an error object if it’s not.
 * @param validation The instance of L10nValidation service
 * @param options A L10n or Intl DateTimeFormatOptions object
 * @param minDate The minimum date
 * @param maxDate The maximum date
 * @param language The current language
 * @return An error object: 'format', 'minDate' or 'maxDate'; null in case the date is valid
 */
export function l10nValidateDate(validation, options, minDate, maxDate, language) {
    const validator = (c) => {
        if (c.value === '' || c.value == null)
            return null;
        const date = validation.parseDate(c.value, options, language);
        if (date != null) {
            if (minDate && date < minDate) {
                return { mindate: true };
            }
            else if (maxDate && date > maxDate) {
                return { maxDate: true };
            }
            return null; // The date is valid.
        }
        else {
            return { format: true };
        }
    };
    return validator;
}
class L10nValidateDateDirective {
    set l10nValidateDate(options) {
        if (options)
            this.options = options;
    }
    constructor(validation) {
        this.validation = validation;
    }
    ngOnInit() {
        this.validator = l10nValidateDate(this.validation, this.options, this.minDate, this.maxDate, this.language);
    }
    ngOnChanges() {
        this.validator = l10nValidateDate(this.validation, this.options, this.minDate, this.maxDate, this.language);
    }
    validate(c) {
        return this.validator ? this.validator(c) : null;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidateDateDirective, deps: [{ token: i1.L10nValidation }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nValidateDateDirective, isStandalone: true, selector: "[l10nValidateDate][ngModel],[l10nValidateDate][formControl],[l10nValidateDate][formControlName]", inputs: { l10nValidateDate: "l10nValidateDate", options: "options", minDate: "minDate", maxDate: "maxDate", language: "language" }, providers: [
            { provide: NG_VALIDATORS, useExisting: forwardRef(() => L10nValidateDateDirective), multi: true }
        ], usesOnChanges: true, ngImport: i0 }); }
}
export { L10nValidateDateDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidateDateDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nValidateDate][ngModel],[l10nValidateDate][formControl],[l10nValidateDate][formControlName]',
                    providers: [
                        { provide: NG_VALIDATORS, useExisting: forwardRef(() => L10nValidateDateDirective), multi: true }
                    ],
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nValidation }]; }, propDecorators: { l10nValidateDate: [{
                type: Input
            }], options: [{
                type: Input
            }], minDate: [{
                type: Input
            }], maxDate: [{
                type: Input
            }], language: [{
                type: Input
            }] } });
