// angular-l10n/esm2022/lib/directives/l10n-translate.directive.mjs
import { Directive, Input } from '@angular/core';
import { L10nDirective } from '../models/l10n-directive';
import * as i0 from "@angular/core";
class L10nTranslateDirective extends L10nDirective {
    set l10nTranslate(params) {
        if (params)
            this.params = params;
    }
    set translate(params) {
        if (params)
            this.params = params;
    }
    getValue(text) {
        return this.translation.translate(text, this.params, this.language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslateDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.0.0", type: L10nTranslateDirective, isStandalone: true, selector: "[l10nTranslate],[translate]", inputs: { l10nTranslate: "l10nTranslate", translate: "translate", params: "params" }, usesInheritance: true, ngImport: i0 }); }
}
export { L10nTranslateDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslateDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[l10nTranslate],[translate]',
                    standalone: true
                }]
        }], propDecorators: { l10nTranslate: [{
                type: Input
            }], translate: [{
                type: Input
            }], params: [{
                type: Input
            }] } });
