// angularx-score/esm2022/lib/shared/components/shared-plugin/shared-plugin.component.mjs
import { Component } from '@angular/core';
import { ConfirmationService, MessageService } from 'primeng/api';
import * as i0 from "@angular/core";
import * as i1 from "angular-l10n";
import * as i2 from "../../services/helper.service";
import * as i3 from "primeng/api";
import * as i4 from "primeng/toast";
export class SharedPluginComponent {
    constructor(translate, helperService, messageService, confirmationService) {
        this.translate = translate;
        this.helperService = helperService;
        this.messageService = messageService;
        this.confirmationService = confirmationService;
        this.helperService.MESSAGE_TOASTER.subscribe(res => {
            if (res) {
                const summary = this.translate.translate('common.toast.messageSummary');
                const detail = this.translate.translate(res.code || 'empty', res.value || { data: '' });
                this.messageService.add({ severity: res.severity, summary, detail });
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: SharedPluginComponent, deps: [{ token: i1.L10nTranslationService }, { token: i2.HelperService }, { token: i3.MessageService }, { token: i3.ConfirmationService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.5", type: SharedPluginComponent, selector: "shared-plugin", providers: [ConfirmationService, MessageService], ngImport: i0, template: "<p-toast position=\"bottom-right\" [baseZIndex]=\"10000\"></p-toast>\n", dependencies: [{ kind: "component", type: i4.Toast, selector: "p-toast", inputs: ["key", "autoZIndex", "baseZIndex", "life", "style", "styleClass", "position", "preventOpenDuplicates", "preventDuplicates", "showTransformOptions", "hideTransformOptions", "showTransitionOptions", "hideTransitionOptions", "breakpoints"], outputs: ["onClose"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: SharedPluginComponent, decorators: [{
            type: Component,
            args: [{ selector: 'shared-plugin', providers: [ConfirmationService, MessageService], template: "<p-toast position=\"bottom-right\" [baseZIndex]=\"10000\"></p-toast>\n" }]
        }], ctorParameters: function () { return [{ type: i1.L10nTranslationService }, { type: i2.HelperService }, { type: i3.MessageService }, { type: i3.ConfirmationService }]; } });
