// angularx-score/esm2022/lib/shared/components/category-selector/category-selector.component.mjs
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { BaseComponent } from '../base-component/base-component.component';
import { BaseControl } from '../../models/base.control';
import * as i0 from "@angular/core";
import * as i1 from "../drop-down/drop-down.component";
export class CategorySelectorComponent extends BaseComponent {
    constructor(_injector, 
    // public categoryService: CategoryService,
    cdr) {
        super(_injector);
        this._injector = _injector;
        this.cdr = cdr;
        // formControl of this input
        this.property = new BaseControl();
        this.typeCode = '';
        this.onChange = new EventEmitter();
        this.catogoryOptions = [];
        this.optionValue = 'id';
    }
    ngAfterViewInit() {
        this.loadCategoryOption();
        if (!!this.fieldValue) {
            this.optionValue = this.fieldValue;
        }
    }
    loadCategoryOption() {
        this.categoryService.findByTypeCode(this.typeCode)
            .subscribe((res) => {
            this.catogoryOptions = res.data;
            this.cdr.detectChanges();
        });
    }
    _onChange(event) {
        this.onChange.emit(event);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: CategorySelectorComponent, deps: [{ token: i0.Injector }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.5", type: CategorySelectorComponent, selector: "app-category-selector", inputs: { property: "property", labelValue: "labelValue", fieldValue: "fieldValue", categoryService: "categoryService", typeCode: "typeCode" }, outputs: { onChange: "onChange" }, usesInheritance: true, ngImport: i0, template: "<app-drop-down [property]=\"property\" [labelValue]=\"labelValue\" [showClear]=\"true\"\n    [options]=\"catogoryOptions\" optionLabel=\"name\" [optionValue]=\"optionValue\" (onChange)=\"_onChange($event)\"></app-drop-down>", styles: [""], dependencies: [{ kind: "component", type: i1.DropDownComponent, selector: "app-drop-down", inputs: ["property", "options", "optionLabel", "optionValue", "labelValue", "showClear"], outputs: ["onChange"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: CategorySelectorComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-category-selector', template: "<app-drop-down [property]=\"property\" [labelValue]=\"labelValue\" [showClear]=\"true\"\n    [options]=\"catogoryOptions\" optionLabel=\"name\" [optionValue]=\"optionValue\" (onChange)=\"_onChange($event)\"></app-drop-down>" }]
        }], ctorParameters: function () { return [{ type: i0.Injector }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { property: [{
                type: Input
            }], labelValue: [{
                type: Input
            }], fieldValue: [{
                type: Input
            }], categoryService: [{
                type: Input
            }], typeCode: [{
                type: Input
            }], onChange: [{
                type: Output
            }] } });
