// angularx-score/esm2022/lib/shared/models/base.control.mjs
import { FormControl } from '@angular/forms';
import { INPUT_TYPE } from '../app-config';
import { CommonUtils } from '../services/common-utils.service';
import { ValidationService } from '../services/validation.service';
export class BaseControl extends FormControl {
    constructor() {
        super(...arguments);
        this.propertyName = '';
        this.actionForm = '';
        this.resource = '';
        this.css = '';
        this.isHide = false;
        this.isRequire = false;
        this.dateFormat = CommonUtils.getDateFormat();
        this.isMultiLanguage = false;
    }
    configBaseControl(propertyConfig, oldValidator, type) {
        if (type) {
            this.type = type;
        }
        // Xu ly set mac dinh required neu khong co cau hinh trong database
        if (oldValidator) {
            if (typeof oldValidator === 'function') {
                const val = oldValidator;
                if (val === ValidationService.required) {
                    this.isRequire = true;
                }
            }
            else {
                for (const index in oldValidator) {
                    const val = oldValidator[index];
                    if (val === ValidationService.required) {
                        this.isRequire = true;
                    }
                }
            }
        }
        if (propertyConfig) {
            this.propertyConfig = propertyConfig;
            this.isHide = this.propertyConfig.isHide;
            this.isRequire = this.propertyConfig.isRequire;
            this.isMultiLanguage = this.propertyConfig.isTranslation;
            this.css = this.propertyConfig.css;
            this.dateFormat = this.propertyConfig.dateFormat ? this.propertyConfig.dateFormat : this.dateFormat;
            this.setType();
        }
        this.listValidation = ValidationService.getValidatorArr(oldValidator, this.propertyConfig);
    }
    getPropertyConfig() {
        return this.propertyConfig;
    }
    getListValidation() {
        return this.listValidation;
    }
    setType() {
        if (this.propertyConfig.moneyFormat) {
            this.type = INPUT_TYPE.CURRENCY;
            return;
        }
        if (this.propertyConfig.numberFormat) {
            this.type = INPUT_TYPE.NUMBER;
            return;
        }
        if (this.propertyConfig.dateFormat) {
            this.type = INPUT_TYPE.DATE;
            return;
        }
        // if (this.propertyConfig.maxLength >= 200) {
        //   this.type = INPUT_TYPE.TEXT_AREA;
        //   return;
        // }
        this.type = INPUT_TYPE.TEXT;
    }
}
