// angularx-score/esm2022/lib/shared/directive/auto-focus.directive.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class AutoFocusDirective {
    constructor(el) {
        this.el = el;
    }
    set autoFocus(value) {
        this._value = value;
    }
    ngAfterViewInit() {
        setTimeout(() => {
            if (this._value) {
                this.el.nativeElement.focus();
            }
        }, 400);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: AutoFocusDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.5", type: AutoFocusDirective, selector: "[autoFocus]", inputs: { autoFocus: "autoFocus" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: AutoFocusDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[autoFocus]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { autoFocus: [{
                type: Input
            }] } });
