// angularx-score/esm2022/lib/shared/directive/drag-drop.directive.mjs
import { Directive, HostBinding, HostListener, Output, EventEmitter } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export class DragDirective {
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.files = new EventEmitter();
        this.background = "#fff";
        this.borderColor = "#E9EAFF";
    }
    onDragOver(evt) {
        evt.preventDefault();
        evt.stopPropagation();
        this.background = "#E9EAFF";
        this.borderColor = "#141ED2";
    }
    onDragLeave(evt) {
        evt.preventDefault();
        evt.stopPropagation();
        this.background = "#fff";
        this.borderColor = "#E9EAFF";
    }
    onDrop(evt) {
        if (!evt || !evt.dataTransfer) {
            return;
        }
        evt.preventDefault();
        evt.stopPropagation();
        this.background = '#fff';
        this.borderColor = "#E9EAFF";
        let files = [];
        for (let i = 0; i < evt.dataTransfer.files.length; i++) {
            const file = evt.dataTransfer.files[i];
            const url = this.sanitizer.bypassSecurityTrustUrl(window.URL.createObjectURL(file));
            files.push({ file, url });
        }
        if (files.length > 0) {
            this.files.emit(files);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: DragDirective, deps: [{ token: i1.DomSanitizer }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.5", type: DragDirective, selector: "[appDrag]", outputs: { files: "files" }, host: { listeners: { "dragover": "onDragOver($event)", "dragleave": "onDragLeave($event)", "drop": "onDrop($event)" }, properties: { "style.background": "this.background", "style.borderColor": "this.borderColor" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: DragDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: "[appDrag]"
                }]
        }], ctorParameters: function () { return [{ type: i1.DomSanitizer }]; }, propDecorators: { files: [{
                type: Output
            }], background: [{
                type: HostBinding,
                args: ["style.background"]
            }], borderColor: [{
                type: HostBinding,
                args: ["style.borderColor"]
            }], onDragOver: [{
                type: HostListener,
                args: ["dragover", ["$event"]]
            }], onDragLeave: [{
                type: HostListener,
                args: ["dragleave", ["$event"]]
            }], onDrop: [{
                type: HostListener,
                args: ['drop', ['$event']]
            }] } });
