// angularx-score/esm2022/lib/shared/directive/has-not-permission.directive.mjs
import { Directive, Input } from '@angular/core';
import { CommonUtils } from '../services/common-utils.service';
import * as i0 from "@angular/core";
export class HasNotPermissionDirective {
    constructor(element, templateRef, viewContainer) {
        this.element = element;
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    set hasNotPermission(value) {
        this._value = value;
        this.updateView(this._value);
    }
    ngOnInit() {
    }
    updateView(value) {
        // if (!value[0].permissions || value[0].permissions.length <= 0) {
        //   this.viewContainer.createEmbeddedView(this.templateRef);
        // } else {
        //   const rsSearch = value[0].permissions.findIndex((x: any) => x.operationCode === CommonUtils.getPermissionCode(value[1]));
        //   if (rsSearch >= 0) {
        //     this.viewContainer.clear();
        //   } else {
        //     this.viewContainer.createEmbeddedView(this.templateRef);
        //   }
        // }
        if (!CommonUtils.havePermission(value[1], value[2])) {
            this.viewContainer.createEmbeddedView(this.templateRef);
        }
        else {
            this.viewContainer.clear();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HasNotPermissionDirective, deps: [{ token: i0.ElementRef }, { token: i0.TemplateRef }, { token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.5", type: HasNotPermissionDirective, selector: "[hasNotPermission]", inputs: { hasNotPermission: "hasNotPermission" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HasNotPermissionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[hasNotPermission]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.TemplateRef }, { type: i0.ViewContainerRef }]; }, propDecorators: { hasNotPermission: [{
                type: Input
            }] } });
