// angularx-score/esm2022/lib/shared/directive/has-permission.directive.mjs
import { Directive, Input } from '@angular/core';
import { CommonUtils } from '../services/common-utils.service';
import * as i0 from "@angular/core";
export class HasPermissionDirective {
    constructor(element, templateRef, viewContainer) {
        this.element = element;
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
    }
    set hasPermission(value) {
        this._value = value;
        this.updateView(this._value);
    }
    ngOnInit() {
    }
    updateView(value) {
        if (value[2]) { //resource key
            if (CommonUtils.havePermission(value[1], value[2])) {
                this.viewContainer.createEmbeddedView(this.templateRef);
            }
            else {
                this.viewContainer.clear();
            }
        }
        else {
            if (!value[0].permissions || value[0].permissions.length <= 0) {
                this.viewContainer.clear();
            }
            else {
                const rsSearch = value[0].permissions.findIndex((x) => x.operationCode === CommonUtils.getPermissionCode(value[1]));
                if (rsSearch < 0) {
                    this.viewContainer.clear();
                }
                else {
                    this.viewContainer.createEmbeddedView(this.templateRef);
                }
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HasPermissionDirective, deps: [{ token: i0.ElementRef }, { token: i0.TemplateRef }, { token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.5", type: HasPermissionDirective, selector: "[hasPermission]", inputs: { hasPermission: "hasPermission" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HasPermissionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[hasPermission]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.TemplateRef }, { type: i0.ViewContainerRef }]; }, propDecorators: { hasPermission: [{
                type: Input
            }] } });
