// angularx-score/esm2022/lib/shared/directive/input-special.directive.mjs
import { NG_VALIDATORS } from '@angular/forms';
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class InputSpecialDirective {
    set type(value) {
        this._type = value;
    }
    set othertype(value) {
        this._othertype = value;
    }
    constructor() {
        this._type = '';
        this._othertype = '';
        this.validator = this.validateSpecialCharecter();
    }
    validate(c) {
        return this.validator(c);
    }
    validateSpecialCharecter() {
        return (c) => {
            const text = c.value;
            if (text && (this._type === 'text' || this._type === 'email' || this._type === 'tel') && !this._othertype) {
                const iChars = '!#$^*[]\\{}\"?<>\'';
                for (let j = 0; j < text.length; j++) {
                    if (iChars.indexOf(text.charAt(j)) >= 0) {
                        return { isValidInput: true };
                    }
                }
            }
            return null;
        };
    }
    registerOnValidatorChange(fn) {
        this._onChange = fn;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: InputSpecialDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.5", type: InputSpecialDirective, selector: "input[type=\"text\"],input[type=\"email\"],input[type=\"tel\"],input:not([othertype=\"url\"]),textarea", inputs: { type: "type", othertype: "othertype" }, providers: [
            {
                provide: NG_VALIDATORS,
                useExisting: InputSpecialDirective,
                multi: true
            }
        ], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: InputSpecialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'input[type="text"],input[type="email"],input[type="tel"],input:not([othertype="url"]),textarea',
                    providers: [
                        {
                            provide: NG_VALIDATORS,
                            useExisting: InputSpecialDirective,
                            multi: true
                        }
                    ]
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { type: [{
                type: Input
            }], othertype: [{
                type: Input
            }] } });
