// angularx-score/esm2022/lib/shared/app-config.mjs
export const APP_CONSTANTS = {};
export const CONFIG_DEFAULT = {
    dateFormat: 'dd/MM/yyyy'
};
export const CONFIG = {
    API_PATH: {
        home: '/v1/home',
        categoryType: '/v1/category-type',
        category: '/v1/category',
        organization: '/v1/organization',
        employee: '/v1/employee',
        educationType: '/v1/education-type',
        trainingCourse: '/v1/training-course',
        course: '/v1/course',
        classroom: '/v1/classroom',
        classroomSchedule: '/v1/classroom-schedule',
        employeeRequest: '/v1/employee-request',
        employeeClassroom: '/v1/employee-classroom',
        employeeClassroomAttend: '/v1/employee-classroom-attend',
        certificateApproval: '/v1/certificate-approval',
        certificateBook: '/v1/certificate-book',
        role: '/v1/role',
        resource: '/v1/resource',
        user: '/v1/user',
        auth: '/v1/auth',
        student: '/v1/student',
        subject: '/v1/subject',
        schoolYear: '/v1/school-year',
        classSubject: '/v1/class-subject',
        studentClassSubject: '/v1/student-class-subject',
    }
};
export const PERMISSION_CODE = {
    // action tac dong
    'action.view': 'VIEW',
    'action.insert': 'INSERT',
    'action.update': 'UPDATE',
    'action.delete': 'DELETE',
    'action.import': 'IMPORT',
    'action.export': 'EXPORT',
    'action.approve': 'APPROVE',
    'action.decide': 'DECIDE',
    'action.unDecide': 'UN_DECIDE',
    'action.unApproveAll': 'UN_APPROVE_ALL',
    'action.approveAll': 'APPROVE_ALL',
    'action.removeEmp': 'REMOVE_EMP',
    'action.addEmp': 'ADD_EMP',
    'action.unApprove': 'UN_APPROVE',
    'action.unLock': 'UN_LOCK',
    'action.lock': 'LOCK',
    'action.calculate': 'CALCULATE',
    'action.viewHistory': 'VIEW_HISTORY',
    'action.enable': 'ENABLE',
    'action.disable': 'DISABLE',
    'action.issueAgain': 'ISSUE_AGAIN',
    'action.issueChange': 'ISSUE_CHANGE',
    'action.quickImport': 'QUICK_IMPORT',
    'action.transfer': 'TRANSFER',
    'action.manage': 'MANAGE',
    'action.viewAll': 'VIEW_ALL',
    'action.payrollCalculate': 'PAYROLL_CALCULATE',
    'action.payrollDestroy': 'DESTROY_PAYROLL',
    'action.syncTax': 'SYNC_TAX',
    'action.syncPayroll': 'SYNC_PAYROLL',
    'action.addAction': 'ADD_ACTION',
    'action.grantRole': 'GRANT_ROLE',
    'action.grantMenu': 'GRANT_MENU',
    'action.grantPermission': 'GRANT_PERMISSION',
    'action.updateStatus': 'UPDATE_STATUS',
    'action.resetPassword': 'RESET_PASSWORD',
    'action.lockUser': 'LOCK_USER',
    'action.brag': 'BRAG' // Dồn túi & làm phách
    ,
    'action.dataSynthesis': 'DATA_SYNTHESIS' // Tổng hợp dữ liệu
    ,
    'action.mngStudent': 'MNG_STUDENT' // Tổng hợp dữ liệu
    ,
    'resource.organization': 'ORGANIZATION',
    'resource.employee': 'EMPLOYEE',
    'resource.trainingCourse': 'TRAINING_COURSE',
    'resource.educationType': 'EDUCATION_TYPE',
    'resource.category': 'CATEGORY',
    'resource.employeeRequest': 'EMPLOYEE_REQUEST',
    'resource.employeeClassroom': 'EMPLOYEE_CLASSROOM',
    'resource.employeeClassroomAttend': 'EMPLOYEE_CLASSROOM_ATTEND',
    'resource.course': 'COURSE',
    'resource.classroom': 'CLASSROOM',
    'resource.classroomSchedule': 'CLASSROOM_SCHEDULE',
    'resource.gradingAssignments': 'GRADING_ASSIGNMENTS' // Chấm bài thu hoạch
    ,
    'resource.certificateApproval': 'CERTIFICATE_APPROVAL' // Xét duyệt cấp chứng chỉ
    ,
    'resource.certificateBook': 'CERTIFICATE_BOOK' // Sổ cấp chứng chỉ
    ,
    'resource.role': 'ROLE',
    'resource.user': 'USER',
    'resource.student': 'STUDENT',
    'resource.subject': 'SUBJECT',
    'resource.schoolYear': 'SCHOOL_YEAR',
    'resource.classSubject': 'CLASS_SUBJECT',
    'resource.roomType': 'ROOM_TYPE',
    'resource.room': 'ROOM',
    'resource.contract': 'CONTRACT',
    'resource.waterElectric': 'WATER_ELECTRIC',
    'resource.waterElectricBill': 'WATER_ELECTRIC_BILL'
};
export var RESOURCE;
(function (RESOURCE) {
    RESOURCE["HOME"] = "HOME";
    RESOURCE["OVERALL_STATISTICS"] = "OVERALL_STATISTICS";
    RESOURCE["SYS_PROPERTY"] = "SYS_PROPERTY";
    RESOURCE["SETTING"] = "SETTING";
    RESOURCE["CATEGORY_TYPE"] = "CATEGORY_TYPE";
    RESOURCE["CATEGORY"] = "CATEGORY";
    // Tổ chức - bộ máy, biên chế
    RESOURCE["MNG_ORGANIZATION"] = "MNG_ORGANIZATION";
    RESOURCE["ORGANIZATION"] = "ORGANIZATION";
    RESOURCE["EMPLOYEE"] = "EMPLOYEE";
    RESOURCE["EDUCATION_TYPE"] = "EDUCATION_TYPE";
    RESOURCE["TRAINING_COURSE"] = "TRAINING_COURSE";
    RESOURCE["COURSE"] = "COURSE";
    RESOURCE["CLASSROOM"] = "CLASSROOM";
    RESOURCE["ROLE"] = "ROLE";
    RESOURCE["USER"] = "USER";
    RESOURCE["AUTH"] = "AUTH";
    RESOURCE["STUDENT"] = "STUDENT";
    RESOURCE["SUBJECT"] = "SUBJECT";
    RESOURCE["SCHOOL_YEAR"] = "SCHOOL_YEAR";
    RESOURCE["CLASS_SUBJECT"] = "CLASS_SUBJECT";
    RESOURCE["STUDENT_CLASS_SUBJECT"] = "STUDENT_CLASS_SUBJECT";
    RESOURCE["ROOM_TYPE"] = "ROOM_TYPE";
    RESOURCE["ROOM"] = "ROOM";
    RESOURCE["CONTRACT"] = "CONTRACT";
    RESOURCE["WATER_ELECTRIC"] = "WATER_ELECTRIC";
    RESOURCE["WATER_ELECTRIC_BILL"] = "WATER_ELECTRIC_BILL";
})(RESOURCE || (RESOURCE = {}));
export var INPUT_TYPE;
(function (INPUT_TYPE) {
    INPUT_TYPE["TEXT"] = "text";
    INPUT_TYPE["NUMBER"] = "number";
    INPUT_TYPE["TEXT_AREA"] = "text-area";
    INPUT_TYPE["CURRENCY"] = "currency";
    INPUT_TYPE["DATE"] = "date";
    INPUT_TYPE["URL"] = "url";
    INPUT_TYPE["PASSWORD"] = "password";
})(INPUT_TYPE || (INPUT_TYPE = {}));
export var ACTION_FORM;
(function (ACTION_FORM) {
    ACTION_FORM["SEARCH"] = "SEARCH";
    ACTION_FORM["INSERT"] = "INSERT";
    ACTION_FORM["UPDATE"] = "UPDATE";
    ACTION_FORM["IMPORT"] = "IMPORT";
    ACTION_FORM["VIEW"] = "VIEW";
    ACTION_FORM["DELETE"] = "DELETE";
    ACTION_FORM["CONFIG"] = "CONFIG";
    ACTION_FORM["UPDATE_STATUS"] = "UPDATE_STATUS";
    ACTION_FORM["SYS_CAT_TYPE_INSERT"] = "SYS_CAT_TYPE_INSERT";
})(ACTION_FORM || (ACTION_FORM = {}));
export var SEARCH_OPERATION;
(function (SEARCH_OPERATION) {
    SEARCH_OPERATION["IN"] = "in";
    SEARCH_OPERATION["GREATER_THAN"] = "gt";
    SEARCH_OPERATION["LESS_THAN"] = "lt";
    SEARCH_OPERATION["GREATER_THAN_EQUAL"] = "gte";
    SEARCH_OPERATION["LESS_THAN_EQUAL"] = "lte";
    SEARCH_OPERATION["NOT_EQUAL"] = "neq";
    SEARCH_OPERATION["EQUAL"] = "eq";
    SEARCH_OPERATION["MATCH"] = "ma";
    SEARCH_OPERATION["MATCH_END"] = "mae";
})(SEARCH_OPERATION || (SEARCH_OPERATION = {}));
export const GENDER_OPTIONS = [
    { label: "Nam", value: 1 },
    { label: "Nữ", value: 0 },
];
export const OBJ_TYPE_OPTIONS = [
    { label: "Cán bộ", value: "OFFICERS" },
    { label: "Học viên", value: "STUDENT" }
];
export const STATUS_DEFENCE_SECURITY = [
    { label: "Chưa học", value: 0 },
    { label: "Đã học", value: 1 },
    { label: "Chưa đủ điều kiện", value: 2 },
    { label: "Đã đăng ký học", value: 3 },
];
export var CLASSROOM_STATUS_ENUM;
(function (CLASSROOM_STATUS_ENUM) {
    CLASSROOM_STATUS_ENUM["TAO_MOI"] = "TAO_MOI";
    CLASSROOM_STATUS_ENUM["DANG_HOC"] = "DANG_HOC";
    CLASSROOM_STATUS_ENUM["DA_THAM_QUAN"] = "DA_THAM_QUAN";
    CLASSROOM_STATUS_ENUM["DA_THI"] = "DA_THI";
    CLASSROOM_STATUS_ENUM["DA_KET_THUC"] = "DA_KET_THUC";
})(CLASSROOM_STATUS_ENUM || (CLASSROOM_STATUS_ENUM = {}));
export const CLASSROOM_STATUS = [
    { label: "Tạo mới", value: CLASSROOM_STATUS_ENUM.TAO_MOI },
    { label: "Đang học", value: CLASSROOM_STATUS_ENUM.DANG_HOC },
    { label: "Đã đi tham quan", value: CLASSROOM_STATUS_ENUM.DA_THAM_QUAN },
    { label: "Đã thi", value: CLASSROOM_STATUS_ENUM.DA_THI },
    { label: "Đã kết thúc", value: CLASSROOM_STATUS_ENUM.DA_KET_THUC },
];
export var CLASSROOM_SCHEDULE_TYPE_ENUM;
(function (CLASSROOM_SCHEDULE_TYPE_ENUM) {
    CLASSROOM_SCHEDULE_TYPE_ENUM["LY_THUYET"] = "LY_THUYET";
    CLASSROOM_SCHEDULE_TYPE_ENUM["THAM_QUAN"] = "THAM_QUAN";
    CLASSROOM_SCHEDULE_TYPE_ENUM["THI"] = "THI";
})(CLASSROOM_SCHEDULE_TYPE_ENUM || (CLASSROOM_SCHEDULE_TYPE_ENUM = {}));
export const CLASSROOM_SCHEDULE_TYPE = [
    { label: "Lý thuyết", value: CLASSROOM_SCHEDULE_TYPE_ENUM.LY_THUYET },
    { label: "Tham quan", value: CLASSROOM_SCHEDULE_TYPE_ENUM.THAM_QUAN },
    { label: "Thi", value: CLASSROOM_SCHEDULE_TYPE_ENUM.THI },
];
