// angularx-score/esm2022/lib/shared/shared.module.mjs
import { NgModule } from '@angular/core';
import { ButtonModule } from 'primeng/button';
import { BreadcrumbModule } from 'primeng/breadcrumb';
import { PanelModule } from 'primeng/panel';
import { TableModule } from 'primeng/table';
import { BaseComponent } from './components/base-component/base-component.component';
import { DialogModule } from 'primeng/dialog';
import { SharedDirectiveModule } from './directive/shared-directive.module';
import { DialogService, DynamicDialogModule } from 'primeng/dynamicdialog';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { OverlayPanelModule } from 'primeng/overlaypanel';
import { InputSpecialDirective } from './directive/input-special.directive';
import { AutoFocusDirective } from './directive/auto-focus.directive';
import { ConfirmDialogModule } from 'primeng/confirmdialog';
import { ControlMessagesComponent } from './components/control-messages/control-messages.component';
import { HttpClientModule } from '@angular/common/http';
import { ToastModule } from 'primeng/toast';
import { TextInputComponent } from './components/text-input/text-input.component';
import { InputTrimDirective } from './directive/input-trim.directive';
import { L10nTranslateAsyncPipe } from 'angular-l10n';
import { DropdownModule } from 'primeng/dropdown';
import { DropDownComponent } from './components/drop-down/drop-down.component';
import { OrgSelectorComponent } from './components/org-selector/org-selector.component';
import { OrgSelectorDialogComponent } from './components/org-selector-dialog/org-selector-dialog.component';
import { DatePickerComponent } from './components/date-picker/date-picker.component';
import { CalendarModule } from 'primeng/calendar';
import { TreeModule } from 'primeng/tree';
import { RadioButtonModule } from 'primeng/radiobutton';
import { ProgressSpinnerModule } from 'primeng/progressspinner';
import { InputSwitchModule } from 'primeng/inputswitch';
import { TabViewModule } from 'primeng/tabview';
import { InputNumberModule } from 'primeng/inputnumber';
import { CheckboxModule } from 'primeng/checkbox';
import { InputTextModule } from 'primeng/inputtext';
import { SplitButtonModule } from 'primeng/splitbutton';
import { ScrollPanelModule } from 'primeng/scrollpanel';
import { MultiSelectModule } from 'primeng/multiselect';
import { OrgTreeComponent } from './components/org-tree/org-tree.component';
import { RadioButtonComponent } from './components/radio-button/radio-button.component';
import { CategorySelectorComponent } from './components/category-selector/category-selector.component';
import { ChooseFileImportDialogComponent } from './components/choose-file-import-dialog/choose-file-import-dialog.component';
import { DragDirective } from './directive/drag-drop.directive';
import { ImportErrorListComponent } from './components/import-error-list/import-error-list.component';
import { InputSwitchComponent } from './components/input-switch/input-switch.component';
import { EmployeePickerComponent } from './components/employee-picker/employee-picker.component';
import { EmployeePickerDialogComponent } from './components/employee-picker-dialog/employee-picker-dialog.component';
import { MultiSelectComponent } from './components/multi-select/multi-select.component';
import { HasPermissionDirective } from './directive/has-permission.directive';
import { HasNotPermissionDirective } from './directive/has-not-permission.directive';
import { ChangePasswordFormComponent } from './components/change-password-form/change-password-form.component';
import { SharedPluginComponent } from './components/shared-plugin/shared-plugin.component';
import * as i0 from "@angular/core";
export class SharedModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: SharedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.5", ngImport: i0, type: SharedModule, declarations: [DragDirective,
            BaseComponent,
            ControlMessagesComponent,
            InputSpecialDirective,
            AutoFocusDirective,
            InputTrimDirective,
            HasPermissionDirective,
            HasNotPermissionDirective,
            SharedPluginComponent,
            TextInputComponent,
            DropDownComponent,
            OrgSelectorComponent,
            OrgSelectorDialogComponent,
            DatePickerComponent,
            OrgTreeComponent,
            RadioButtonComponent,
            CategorySelectorComponent,
            ChooseFileImportDialogComponent,
            ImportErrorListComponent,
            InputSwitchComponent,
            EmployeePickerComponent,
            EmployeePickerDialogComponent,
            MultiSelectComponent,
            ChangePasswordFormComponent], imports: [CommonModule,
            ButtonModule,
            BreadcrumbModule,
            PanelModule,
            TableModule,
            DialogModule,
            SharedDirectiveModule,
            DynamicDialogModule,
            FormsModule,
            ReactiveFormsModule,
            OverlayPanelModule,
            ConfirmDialogModule,
            HttpClientModule,
            ToastModule,
            L10nTranslateAsyncPipe,
            DropdownModule,
            CalendarModule,
            TreeModule,
            RadioButtonModule,
            ProgressSpinnerModule,
            InputSwitchModule,
            TabViewModule,
            InputNumberModule,
            CheckboxModule,
            InputTextModule,
            SplitButtonModule,
            ScrollPanelModule,
            MultiSelectModule,
            ButtonModule], exports: [ButtonModule,
            BreadcrumbModule,
            PanelModule,
            TableModule,
            DialogModule,
            SharedDirectiveModule,
            DynamicDialogModule,
            FormsModule,
            ReactiveFormsModule,
            OverlayPanelModule,
            InputSpecialDirective,
            AutoFocusDirective,
            InputTrimDirective,
            HasPermissionDirective,
            HasNotPermissionDirective,
            ConfirmDialogModule,
            ControlMessagesComponent,
            HttpClientModule,
            ToastModule,
            SharedPluginComponent,
            TextInputComponent,
            L10nTranslateAsyncPipe,
            DropdownModule,
            DropDownComponent,
            OrgSelectorComponent,
            DatePickerComponent,
            CalendarModule,
            TreeModule,
            ButtonModule,
            RadioButtonModule,
            ProgressSpinnerModule,
            InputSwitchModule,
            TabViewModule,
            InputNumberModule,
            CheckboxModule,
            InputTextModule,
            SplitButtonModule,
            ScrollPanelModule,
            MultiSelectModule,
            OrgTreeComponent,
            RadioButtonComponent,
            CategorySelectorComponent,
            InputSwitchComponent,
            EmployeePickerComponent,
            MultiSelectComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: SharedModule, providers: [
            DialogService
        ], imports: [CommonModule,
            ButtonModule,
            BreadcrumbModule,
            PanelModule,
            TableModule,
            DialogModule,
            SharedDirectiveModule,
            DynamicDialogModule,
            FormsModule,
            ReactiveFormsModule,
            OverlayPanelModule,
            ConfirmDialogModule,
            HttpClientModule,
            ToastModule,
            DropdownModule,
            CalendarModule,
            TreeModule,
            RadioButtonModule,
            ProgressSpinnerModule,
            InputSwitchModule,
            TabViewModule,
            InputNumberModule,
            CheckboxModule,
            InputTextModule,
            SplitButtonModule,
            ScrollPanelModule,
            MultiSelectModule,
            ButtonModule, ButtonModule,
            BreadcrumbModule,
            PanelModule,
            TableModule,
            DialogModule,
            SharedDirectiveModule,
            DynamicDialogModule,
            FormsModule,
            ReactiveFormsModule,
            OverlayPanelModule,
            ConfirmDialogModule,
            HttpClientModule,
            ToastModule,
            DropdownModule,
            CalendarModule,
            TreeModule,
            ButtonModule,
            RadioButtonModule,
            ProgressSpinnerModule,
            InputSwitchModule,
            TabViewModule,
            InputNumberModule,
            CheckboxModule,
            InputTextModule,
            SplitButtonModule,
            ScrollPanelModule,
            MultiSelectModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: SharedModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        DragDirective,
                        BaseComponent,
                        ControlMessagesComponent,
                        InputSpecialDirective,
                        AutoFocusDirective,
                        InputTrimDirective,
                        HasPermissionDirective,
                        HasNotPermissionDirective,
                        SharedPluginComponent,
                        TextInputComponent,
                        DropDownComponent,
                        OrgSelectorComponent,
                        OrgSelectorDialogComponent,
                        DatePickerComponent,
                        OrgTreeComponent,
                        RadioButtonComponent,
                        CategorySelectorComponent,
                        ChooseFileImportDialogComponent,
                        ImportErrorListComponent,
                        InputSwitchComponent,
                        EmployeePickerComponent,
                        EmployeePickerDialogComponent,
                        MultiSelectComponent,
                        ChangePasswordFormComponent,
                    ],
                    imports: [
                        CommonModule,
                        ButtonModule,
                        BreadcrumbModule,
                        PanelModule,
                        TableModule,
                        DialogModule,
                        SharedDirectiveModule,
                        DynamicDialogModule,
                        FormsModule,
                        ReactiveFormsModule,
                        OverlayPanelModule,
                        ConfirmDialogModule,
                        HttpClientModule,
                        ToastModule,
                        L10nTranslateAsyncPipe,
                        DropdownModule,
                        CalendarModule,
                        TreeModule,
                        RadioButtonModule,
                        ProgressSpinnerModule,
                        InputSwitchModule,
                        TabViewModule,
                        InputNumberModule,
                        CheckboxModule,
                        InputTextModule,
                        SplitButtonModule,
                        ScrollPanelModule,
                        MultiSelectModule,
                        ButtonModule,
                    ],
                    exports: [
                        ButtonModule,
                        BreadcrumbModule,
                        PanelModule,
                        TableModule,
                        DialogModule,
                        SharedDirectiveModule,
                        DynamicDialogModule,
                        FormsModule,
                        ReactiveFormsModule,
                        OverlayPanelModule,
                        InputSpecialDirective,
                        AutoFocusDirective,
                        InputTrimDirective,
                        HasPermissionDirective,
                        HasNotPermissionDirective,
                        ConfirmDialogModule,
                        ControlMessagesComponent,
                        HttpClientModule,
                        ToastModule,
                        SharedPluginComponent,
                        TextInputComponent,
                        L10nTranslateAsyncPipe,
                        DropdownModule,
                        DropDownComponent,
                        OrgSelectorComponent,
                        DatePickerComponent,
                        CalendarModule,
                        TreeModule,
                        ButtonModule,
                        RadioButtonModule,
                        ProgressSpinnerModule,
                        InputSwitchModule,
                        TabViewModule,
                        InputNumberModule,
                        CheckboxModule,
                        InputTextModule,
                        SplitButtonModule,
                        ScrollPanelModule,
                        MultiSelectModule,
                        OrgTreeComponent,
                        RadioButtonComponent,
                        CategorySelectorComponent,
                        InputSwitchComponent,
                        EmployeePickerComponent,
                        MultiSelectComponent
                    ],
                    providers: [
                        DialogService
                    ]
                }]
        }] });
