// angular-l10n/esm2022/lib/modules/l10n-intl.module.mjs
import { NgModule } from '@angular/core';
import { L10nDatePipe, L10nDateAsyncPipe } from '../pipes/l10n-date.pipe';
import { L10nNumberPipe, L10nNumberAsyncPipe } from '../pipes/l10n-number.pipe';
import { L10nTimeAgoPipe, L10nTimeAgoAsyncPipe } from '../pipes/l10n-time-ago.pipe';
import { L10nPluralPipe, L10nPluralAsyncPipe } from '../pipes/l10n-plural.pipe';
import { L10nDisplayNamesPipe, L10nDisplayNamesAsyncPipe } from '../pipes/l10n-display-names.pipe';
import { L10nDateDirective } from '../directives/l10n-date.directive';
import { L10nNumberDirective } from '../directives/l10n-number.directive';
import { L10nTimeAgoDirective } from '../directives/l10n-time-ago.directive';
import { L10nPluralDirective } from '../directives/l10n-plural.directive';
import { L10nDisplayNamesDirective } from '../directives/l10n-display-names.directive';
import { L10nIntlService } from '../services/l10n-intl.service';
import * as i0 from "@angular/core";
class L10nIntlModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nIntlModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nIntlModule, imports: [L10nDatePipe,
            L10nNumberPipe,
            L10nTimeAgoPipe,
            L10nPluralPipe,
            L10nDisplayNamesPipe,
            L10nDateAsyncPipe,
            L10nNumberAsyncPipe,
            L10nTimeAgoAsyncPipe,
            L10nPluralAsyncPipe,
            L10nDisplayNamesAsyncPipe,
            L10nDateDirective,
            L10nNumberDirective,
            L10nTimeAgoDirective,
            L10nPluralDirective,
            L10nDisplayNamesDirective], exports: [L10nDatePipe,
            L10nNumberPipe,
            L10nTimeAgoPipe,
            L10nPluralPipe,
            L10nDisplayNamesPipe,
            L10nDateAsyncPipe,
            L10nNumberAsyncPipe,
            L10nTimeAgoAsyncPipe,
            L10nPluralAsyncPipe,
            L10nDisplayNamesAsyncPipe,
            L10nDateDirective,
            L10nNumberDirective,
            L10nTimeAgoDirective,
            L10nPluralDirective,
            L10nDisplayNamesDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nIntlModule, providers: [L10nIntlService] }); }
}
export { L10nIntlModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nIntlModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        L10nDatePipe,
                        L10nNumberPipe,
                        L10nTimeAgoPipe,
                        L10nPluralPipe,
                        L10nDisplayNamesPipe,
                        L10nDateAsyncPipe,
                        L10nNumberAsyncPipe,
                        L10nTimeAgoAsyncPipe,
                        L10nPluralAsyncPipe,
                        L10nDisplayNamesAsyncPipe,
                        L10nDateDirective,
                        L10nNumberDirective,
                        L10nTimeAgoDirective,
                        L10nPluralDirective,
                        L10nDisplayNamesDirective
                    ],
                    exports: [
                        L10nDatePipe,
                        L10nNumberPipe,
                        L10nTimeAgoPipe,
                        L10nPluralPipe,
                        L10nDisplayNamesPipe,
                        L10nDateAsyncPipe,
                        L10nNumberAsyncPipe,
                        L10nTimeAgoAsyncPipe,
                        L10nPluralAsyncPipe,
                        L10nDisplayNamesAsyncPipe,
                        L10nDateDirective,
                        L10nNumberDirective,
                        L10nTimeAgoDirective,
                        L10nPluralDirective,
                        L10nDisplayNamesDirective
                    ],
                    providers: [L10nIntlService]
                }]
        }] });
