// angular-l10n/esm2022/lib/modules/l10n-validation.module.mjs
import { NgModule } from '@angular/core';
import { L10nValidateNumberDirective } from '../directives/l10n-validate-number.directive';
import { L10nValidateDateDirective } from '../directives/l10n-validate-date.directive';
import { L10nValidation, L10nDefaultValidation } from '../services/l10n-validation';
import * as i0 from "@angular/core";
class L10nValidationModule {
    static forRoot(token = {}) {
        return {
            ngModule: L10nValidationModule,
            providers: [
                { provide: L10nValidation, useClass: token.validation || L10nDefaultValidation }
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nValidationModule, imports: [L10nValidateNumberDirective,
            L10nValidateDateDirective], exports: [L10nValidateNumberDirective,
            L10nValidateDateDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidationModule }); }
}
export { L10nValidationModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidationModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        L10nValidateNumberDirective,
                        L10nValidateDateDirective
                    ],
                    exports: [
                        L10nValidateNumberDirective,
                        L10nValidateDateDirective
                    ]
                }]
        }] });
