// angularx-score/esm2022/lib/shared/components/control-messages/control-messages.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "angular-l10n";
import * as i2 from "@angular/common";
// import { TranslationService } from 'angular-l10n';
export class ControlMessagesComponent {
    constructor(translation) {
        this.translation = translation;
        // tslint:disable-next-line: max-line-length
        this.replaceKeys = ['max', 'min', 'maxlength', 'minlength', 'dateNotAffter', 'dateNotBefore', 'duplicateArray', 'beforeCurrentDate', 'required', 'numberNotGreater'];
        // tslint:disable-next-line: max-line-length
        this.actualKeys = ['max', 'min', 'requiredLength', 'requiredLength', 'dateNotAffter', 'dateNotBefore', 'duplicateArray', 'beforeCurrentDate', 'required', 'numberNotGreater'];
        this.translateKeys = ['dateNotAffter', 'dateNotBefore', 'cannotContainSpace', 'duplicateArray', 'beforeCurrentDate', 'required', 'numberNotGreater'];
    }
    get errorMessage() {
        for (const propertyName in this.control.errors) {
            if (this.control.errors.hasOwnProperty(propertyName) && this.control.touched) {
                const messageText = this.translation.translate(`validate.${propertyName}`);
                // const messageText = `validate.${propertyName}`;
                const errors = this.control.errors[propertyName];
                return this.buildMessage(messageText, errors);
            }
        }
        return '';
    }
    markAsUntouched() {
        this.control.markAsUntouched();
    }
    /**
     * buildMessage
     * @param messageText: string
     * @param errors: ValidationErrors
     */
    buildMessage(messageText, errors) {
        for (const i in this.replaceKeys) {
            if (errors && errors.hasOwnProperty(this.actualKeys[i])) {
                let text = errors[this.actualKeys[i]];
                if (this.translateKeys.indexOf(this.actualKeys[i]) !== -1) {
                    text = this.translation.translate(text);
                    // text = text;
                }
                messageText = messageText.replace(new RegExp('\\$\\{' + this.replaceKeys[i] + '\\}', 'g'), text);
            }
        }
        return messageText;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: ControlMessagesComponent, deps: [{ token: i1.L10nTranslationService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.5", type: ControlMessagesComponent, selector: "app-control-messages", inputs: { control: "control", labelName: "labelName" }, ngImport: i0, template: "<div class=\"errorMessageDiv\" [ngClass]=\"{'show': errorMessage !== undefined}\">\r\n  <small *ngIf=\"errorMessage !== undefined\" (click)=\"markAsUntouched()\" class=\"text-danger errorMessage\">\r\n    {{ errorMessage }}\r\n  </small>\r\n</div>\r\n", styles: ["::ng-deep app-control-messages .errorMessage{color:#a94442!important;margin-top:1rem}\n"], dependencies: [{ kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: ControlMessagesComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-control-messages', template: "<div class=\"errorMessageDiv\" [ngClass]=\"{'show': errorMessage !== undefined}\">\r\n  <small *ngIf=\"errorMessage !== undefined\" (click)=\"markAsUntouched()\" class=\"text-danger errorMessage\">\r\n    {{ errorMessage }}\r\n  </small>\r\n</div>\r\n", styles: ["::ng-deep app-control-messages .errorMessage{color:#a94442!important;margin-top:1rem}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.L10nTranslationService }]; }, propDecorators: { control: [{
                type: Input
            }], labelName: [{
                type: Input
            }] } });
