// angularx-score/esm2022/lib/shared/components/base-component/base-component.component.mjs
import { Component, Injectable, Inject, inject } from '@angular/core';
import { CommonUtils } from '../../services/common-utils.service';
import { ConfirmationService } from 'primeng/api';
import { L10N_LOCALE } from 'angular-l10n';
import { ACTION_FORM, INPUT_TYPE } from '../../app-config';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../../app-config";
export class BaseComponent {
    constructor(injector, actr, _resource, actionForm) {
        this.injector = injector;
        this.actr = actr;
        this._resource = _resource;
        this.actionForm = ACTION_FORM.VIEW;
        this.resource = '';
        this.propertyConfigs = new Array();
        this.locale = inject(L10N_LOCALE);
        this.INPUT_TYPE = INPUT_TYPE;
        this.confirmationService = injector.get(ConfirmationService);
        this.commonUtil = injector.get(CommonUtils);
        if (actionForm) {
            this.actionForm = actionForm;
        }
        if (_resource) {
            this.resource = _resource;
        }
        // this.findPropertyDetails();
        // this.getPermissions();
    }
    /**
     * Build FormGroup
     * @param formData value of controls (Ex: data)
     * @param formConfig object formConfig (Ex: formConfig)
     * @param actionForm action of this Form (Ex: ACTION_FORM.INSERT)
     * @param validateForm validate of FormGroup
     */
    buildForm(formData, formConfig, actionForm, validateForm) {
        if (actionForm) {
            this.actionForm = actionForm;
            // console.log('ActionForm is being built ->', this.actionForm);
        }
        return CommonUtils.createFormNew(this.resource, this.actionForm, formData, formConfig, this.propertyConfigs, validateForm);
    }
    /**
     * confirmMessage
     */
    confirmMessage(messageCode, accept, reject) {
        const message = messageCode || 'Confirm Message';
        const header = 'Xác nhận';
        const btnSave = 'Đồng ý';
        const btnCancel = 'Hủy';
        this.confirmationService.confirm({
            message: message,
            header: header,
            icon: 'pi pi-exclamation-triangle',
            accept: accept,
            reject: reject,
            acceptLabel: btnSave,
            rejectLabel: btnCancel
        });
    }
    requestIsSuccess(res) {
        return res && res.type == 'SUCCESS';
    }
    havePermission(action, resource) {
        return CommonUtils.havePermission(action, resource);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: BaseComponent, deps: [{ token: i0.Injector }, { token: i1.ActivatedRoute }, { token: 'isFactory' }, { token: i2.ACTION_FORM }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.5", type: BaseComponent, selector: "app-base-component", ngImport: i0, template: "<p>base-component works!</p>\n" }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: BaseComponent }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: BaseComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-base-component', template: "<p>base-component works!</p>\n" }]
        }, {
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.Injector }, { type: i1.ActivatedRoute }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['isFactory']
                }] }, { type: i2.ACTION_FORM }]; } });
