// angular-l10n/esm2022/lib/models/l10n-async-pipe.mjs
import { Injectable, ChangeDetectorRef, inject } from '@angular/core';
import { L10nTranslationService } from '../services/l10n-translation.service';
import * as i0 from "@angular/core";
class L10nAsyncPipe {
    constructor() {
        this.translation = inject(L10nTranslationService);
        this.cdr = inject(ChangeDetectorRef);
        this.onChanges = this.translation.onChange().subscribe({
            next: () => this.cdr.markForCheck()
        });
    }
    ngOnDestroy() {
        if (this.onChanges)
            this.onChanges.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nAsyncPipe, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nAsyncPipe }); }
}
export { L10nAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nAsyncPipe, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return []; } });
