// angular-l10n/esm2022/lib/services/l10n-validation.mjs
import { Injectable, Inject } from '@angular/core';
import { L10N_LOCALE } from '../models/l10n-config';
import * as i0 from "@angular/core";
/**
 * Implement this class-interface to create a validation service.
 */
class L10nValidation {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidation, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidation }); }
}
export { L10nValidation };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nValidation, decorators: [{
            type: Injectable
        }] });
class L10nDefaultValidation {
    constructor(locale) {
        this.locale = locale;
    }
    parseNumber(value, options, language = this.locale.numberLanguage || this.locale.language) {
        return null;
    }
    parseDate(value, options, language = this.locale.dateLanguage || this.locale.language) {
        return null;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultValidation, deps: [{ token: L10N_LOCALE }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultValidation }); }
}
export { L10nDefaultValidation };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultValidation, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [L10N_LOCALE]
                }] }]; } });
