// angular-l10n/esm2022/lib/services/l10n-translation-fallback.mjs
import { Injectable, Inject } from '@angular/core';
import { L10N_CONFIG } from '../models/l10n-config';
import * as i0 from "@angular/core";
import * as i1 from "./l10n-cache";
import * as i2 from "./l10n-translation-loader";
/**
 * Implement this class-interface to create a translation fallback.
 */
class L10nTranslationFallback {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslationFallback, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslationFallback }); }
}
export { L10nTranslationFallback };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslationFallback, decorators: [{
            type: Injectable
        }] });
class L10nDefaultTranslationFallback {
    constructor(config, cache, translationLoader) {
        this.config = config;
        this.cache = cache;
        this.translationLoader = translationLoader;
    }
    /**
     * Translation data will be merged in the following order:
     * 'language'
     * 'language[-script]'
     * 'language[-script][-region]'
     */
    get(language, provider) {
        const loaders = [];
        const keywords = language.match(/-?[a-zA-z]+/g) || [];
        let fallbackLanguage = '';
        for (const keyword of keywords) {
            fallbackLanguage += keyword;
            if (this.config.cache) {
                loaders.push(this.cache.read(`${provider.name}-${fallbackLanguage}`, this.translationLoader.get(fallbackLanguage, provider)));
            }
            else {
                loaders.push(this.translationLoader.get(fallbackLanguage, provider));
            }
        }
        return loaders;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultTranslationFallback, deps: [{ token: L10N_CONFIG }, { token: i1.L10nCache }, { token: i2.L10nTranslationLoader }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultTranslationFallback }); }
}
export { L10nDefaultTranslationFallback };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDefaultTranslationFallback, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [L10N_CONFIG]
                }] }, { type: i1.L10nCache }, { type: i2.L10nTranslationLoader }]; } });
