// angular-l10n/esm2022/lib/pipes/l10n-display-names.pipe.mjs
import { Pipe } from '@angular/core';
import { L10nAsyncPipe } from '../models/l10n-async-pipe';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nDisplayNamesPipe {
    constructor(intl) {
        this.intl = intl;
    }
    transform(value, language, options) {
        if (value == null || value === '')
            return null;
        return this.intl.displayNames(value, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesPipe, isStandalone: true, name: "l10nDisplayNames" }); }
}
export { L10nDisplayNamesPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nDisplayNames',
                    pure: true,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
class L10nDisplayNamesAsyncPipe extends L10nAsyncPipe {
    constructor(intl) {
        super();
        this.intl = intl;
    }
    transform(value, options, language) {
        if (value == null || value === '')
            return null;
        return this.intl.displayNames(value, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesAsyncPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesAsyncPipe, isStandalone: true, name: "l10nDisplayNamesAsync", pure: false }); }
}
export { L10nDisplayNamesAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nDisplayNamesAsyncPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nDisplayNamesAsync',
                    pure: false,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
