// angular-l10n/esm2022/lib/pipes/l10n-number.pipe.mjs
import { Pipe } from '@angular/core';
import { L10nAsyncPipe } from '../models/l10n-async-pipe';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nNumberPipe {
    constructor(intl) {
        this.intl = intl;
    }
    transform(value, language, options, currency, convert, convertParams) {
        if (value == null || value === '')
            return null;
        return this.intl.formatNumber(value, options, language, currency, convert, convertParams);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberPipe, isStandalone: true, name: "l10nNumber" }); }
}
export { L10nNumberPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nNumber',
                    pure: true,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
class L10nNumberAsyncPipe extends L10nAsyncPipe {
    constructor(intl) {
        super();
        this.intl = intl;
    }
    transform(value, options, convert, convertParams, language, currency) {
        if (value == null || value === '')
            return null;
        return this.intl.formatNumber(value, options, language, currency, convert, convertParams);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberAsyncPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberAsyncPipe, isStandalone: true, name: "l10nNumberAsync", pure: false }); }
}
export { L10nNumberAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nNumberAsyncPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nNumberAsync',
                    pure: false,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
