// angular-l10n/esm2022/lib/pipes/l10n-time-ago.pipe.mjs
import { Pipe } from '@angular/core';
import { L10nAsyncPipe } from '../models/l10n-async-pipe';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nTimeAgoPipe {
    constructor(intl) {
        this.intl = intl;
    }
    transform(value, language, unit, options) {
        if (value == null || value === '')
            return null;
        return this.intl.formatRelativeTime(value, unit, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoPipe, isStandalone: true, name: "l10nTimeAgo" }); }
}
export { L10nTimeAgoPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nTimeAgo',
                    pure: true,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
class L10nTimeAgoAsyncPipe extends L10nAsyncPipe {
    constructor(intl) {
        super();
        this.intl = intl;
    }
    transform(value, unit, options, language) {
        if (value == null || value === '')
            return null;
        return this.intl.formatRelativeTime(value, unit, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoAsyncPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoAsyncPipe, isStandalone: true, name: "l10nTimeAgoAsync", pure: false }); }
}
export { L10nTimeAgoAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTimeAgoAsyncPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nTimeAgoAsync',
                    pure: false,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
