// angular-l10n/esm2022/lib/pipes/l10n-translate.pipe.mjs
import { Pipe } from '@angular/core';
import { L10nAsyncPipe } from '../models/l10n-async-pipe';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-translation.service";
class L10nTranslatePipe {
    constructor(translation) {
        this.translation = translation;
    }
    transform(key, language, params) {
        if (key == null || key === '')
            return null;
        return this.translation.translate(key, params, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslatePipe, deps: [{ token: i1.L10nTranslationService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslatePipe, isStandalone: true, name: "translate" }); }
}
export { L10nTranslatePipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslatePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'translate',
                    pure: true,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nTranslationService }]; } });
class L10nTranslateAsyncPipe extends L10nAsyncPipe {
    transform(key, params, language) {
        if (key == null || key === '')
            return null;
        return this.translation.translate(key, params, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslateAsyncPipe, deps: null, target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslateAsyncPipe, isStandalone: true, name: "translateAsync", pure: false }); }
}
export { L10nTranslateAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nTranslateAsyncPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'translateAsync',
                    pure: false,
                    standalone: true
                }]
        }] });
