// angular-l10n/esm2022/lib/pipes/l10n-plural.pipe.mjs
import { Pipe } from '@angular/core';
import { L10nAsyncPipe } from '../models/l10n-async-pipe';
import * as i0 from "@angular/core";
import * as i1 from "../services/l10n-intl.service";
class L10nPluralPipe {
    constructor(intl) {
        this.intl = intl;
    }
    transform(value, language, prefix, options) {
        if (value == null || value === '')
            return null;
        return this.intl.plural(value, prefix, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralPipe, isStandalone: true, name: "l10nPlural" }); }
}
export { L10nPluralPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nPlural',
                    pure: true,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
class L10nPluralAsyncPipe extends L10nAsyncPipe {
    constructor(intl) {
        super();
        this.intl = intl;
    }
    transform(value, prefix, options, language) {
        if (value == null || value === '')
            return null;
        return this.intl.plural(value, prefix, options, language);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralAsyncPipe, deps: [{ token: i1.L10nIntlService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralAsyncPipe, isStandalone: true, name: "l10nPluralAsync", pure: false }); }
}
export { L10nPluralAsyncPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.0", ngImport: i0, type: L10nPluralAsyncPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'l10nPluralAsync',
                    pure: false,
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i1.L10nIntlService }]; } });
