// angularx-score/esm2022/lib/shared/services/crypto.service.mjs
import { Injectable } from '@angular/core';
import * as CryptoJS from 'crypto-js';
import * as i0 from "@angular/core";
export class CryptoService {
    static { this.encrText = 'rNZSYvtgfyUPx75Okf6ArEx2SiktAW9j'; }
    static encr(data) {
        try {
            return CryptoJS.AES.encrypt(JSON.stringify(data), this.encrText).toString();
        }
        catch (e) {
            console.log(e);
            return '';
        }
    }
    static decr(data) {
        try {
            const bytes = CryptoJS.AES.decrypt(data, this.encrText);
            if (bytes) {
                return JSON.parse(bytes.toString(CryptoJS.enc.Utf8));
            }
            return data;
        }
        catch (e) {
            console.log(e);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: CryptoService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: CryptoService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: CryptoService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
