// angularx-score/esm2022/lib/shared/services/helper.service.mjs
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
export class HelperService {
    constructor() {
        this.ORGANIZATION = new BehaviorSubject([]);
        this.APP_TOAST_MESSAGE = new BehaviorSubject([]);
        this.APP_CONFIRM_DELETE = new BehaviorSubject(null);
        this.APP_SHOW_HIDE_LEFT_MENU = new BehaviorSubject([]);
        this.APP_SHOW_PROCESSING = new BehaviorSubject([]);
        this.EMPLOYEE_URL = new BehaviorSubject([]);
        this.INSERT_WORK_PROCESS = new BehaviorSubject([]);
        this.CHECK_WORK_PROCESS = new BehaviorSubject([]);
        this.LANGUAGE_CHANGE = new BehaviorSubject(null);
        this.PROCESS_LOADER = new BehaviorSubject(null);
        this.MESSAGE_TOASTER = new BehaviorSubject(null);
        this.CONFIRM = new BehaviorSubject(null);
        this.LOAD_USER = new BehaviorSubject(null);
    }
    /**
     * afterSaveOrganization
     * param data
     */
    afterSaveOrganization(data) {
        this.ORGANIZATION.next(data);
    }
    /**
     * createMessage
     * param data
     */
    processReturnMessage(data) {
        this.APP_TOAST_MESSAGE.next(data);
    }
    /**
     * processing
     * param data
     */
    isProcessing(isProcessing) {
        this.APP_SHOW_PROCESSING.next(isProcessing);
    }
    /**
     * confirmDelete
     * param data
     */
    confirmDelete(data) {
        this.APP_CONFIRM_DELETE.next(data);
        // this.APP_CONFIRM_DELETE.pipe(data);
    }
    refreshConfirmDelete() {
        this.APP_CONFIRM_DELETE = new BehaviorSubject(null);
    }
    /**
     * Load lại cây đơn vị
     */
    reloadTreeOrganization(data) {
        this.ORGANIZATION.next({ type: 'RELOAD_TREE', data: data });
    }
    reloadGridOrganization(data) {
        this.ORGANIZATION.next({ type: 'RELOAD_GRID', data: data });
    }
    resolveUrlEmployee(data) {
        this.EMPLOYEE_URL.next(data);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HelperService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HelperService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: HelperService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return []; } });
