// angularx-score/esm2022/lib/shared/services/import-form.service.mjs
import { Injectable } from "@angular/core";
import { BaseService } from "./basic.service";
import * as i0 from "@angular/core";
export class ImportFormService extends BaseService {
    constructor(_injector) {
        super(_injector, '', '');
        this._injector = _injector;
    }
    downloadTemplate(service, urlEndpoint, extraValue) {
        const url = `${service.getBaseUrl()}/${urlEndpoint}`;
        // return this.getRequest(url);
        const buildParams = this.commonUtil.buildParams(extraValue || {});
        return this.getRequest(url, { params: buildParams, responseType: 'blob' });
    }
    //   public doDownloadFileByName(fileName: string) {
    //     const url = UrlConstant.API_VERSION + '/download/temp-file?fileName=' + fileName;
    //     return this.getRequestFile(url);
    //   }
    doImport(service, endpoint, data) {
        const url = `${service.getBaseUrl()}/${endpoint}`;
        return this.postRequest(url, data);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: ImportFormService, deps: [{ token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: ImportFormService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.5", ngImport: i0, type: ImportFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i0.Injector }]; } });
``;
